packages/authentication/src/__tests__/acceptance/basic-auth.acceptance.ts

```diff
@@ -87,10 +83,36 @@ describe.skip('Basic Authentication', () => {
     });
   }
 
+  // Since it has to be user's job to provide the `verify` function and
+  // instantiate the passport strategy, we cannot add the imported `BasicStrategy`
+  // class as extension directly, we need to wrap it as a strategy provider,
+  // then add the provider class as the extension.
+  // See Line 89 in the function `givenAServer`
```

Comment:
Since we are not able to register a Passport strategy class directly and have to create a custom provider class, I would like to propose a slightly different user experience that does not require any new extension points to be created.

Usage:

```ts
import {Strategy, PassportStrategyAdapter} from '@loopback/authentication';

class PassportBasicAuthProvider implements Provider<Strategy> {
    value(): Strategy {
      // create an instance of the Passport Strategy
      // configure it with "verify" and any other options/behavior as needed
      const passportStrategy = new BasicStrategy(this.verify.bind(this));

      // wrap the Passport Strategy class instance and 
      // adapt it to match LB Strategy interface contract
      return new PassportStrategyAdapter(passportStrategy);
    }

  verify(username: string, password: string, cb: Function) {
    process.nextTick(() => {
      users.find(username, password, cb);
    });
  }
}

// register PassportBasicAuthProvider as an extension
// for AUTHENTICATION_STRATEGY_EXTENSION_POINT_NAME
// (a regular LB4 Authentication strategy)
```

Implementation wise, I think we just need to export existing `StrategyAdapter` class for public consumption, preferably under a more descriptive name. In my example, I am using the name `PassportStrategyAdapter`. (In the future, we can implement additional adapters. E.g. `Auth0StrategyAdapter` (see https://auth0.com, I am totally making this example up.)

An important feature of this approach is extensibility. When each strategy contract (LB4, Passport, etc.) requires a new extension point, then it's difficult to implement support for new strategy contracts in 3rd-party modules. With the proposed Adapter approach, the core authentication framework does not need to deal with different strategy contracts and thus adding a new strategy contract does not require any changes in the core.

In fact, I think we should move `PassportStrategyAdapter` into a standalone package:
 - It will validate extensibility of our design and enforce proper separation of concerns.
 - People using only LB strategies should not have to install `passport` dependency. By moving the passport adapter to a new package, only consumers of this new package will depend on `passport`.
 - As we learn more about Passport, we may need to introduce breaking change in the passport adapter. Ideally, these breaking changes should affect only Passport users, they should not trigger a semver-major release of the entire authentication framework.

/cc @raymondfeng

Reply:
> Since we are not able to register a Passport strategy class directly and have to create a custom provider class, I would like to propose a slightly different user experience that does not require any new extension points to be created.


Neat 👍 yeah since the adapter implements `AuthenticationStrategy` it makes sense to only have one extension. And this could also be a good practice for other adapters like you said in 
> we can implement additional adapters, e.g. Auth0StrategyAdapter (see https://auth0.com, I am totally making this example up.)

Reply:
>  With the proposed Adapter approach, the core authentication framework does not need to deal with different strategy contracts and thus adding a new strategy contract does not require any changes in the core.

+1 Fair enough.